src/CalculatorUnitTests/RationalTest.cpp

```diff
@@ -0,0 +1,226 @@
+            res = Rational(-124) % Rational(-124);
+            VERIFY_ARE_EQUAL(res, 0);
+            res = Rational(24) % Rational(24);
+            VERIFY_ARE_EQUAL(res, 0);
+        }
+
+        TEST_METHOD(TestRemainerZero)
+        {
+            // Test with Zero
+            auto res = Rational(0) % Rational(3654);
+            VERIFY_ARE_EQUAL(res, 0);
+            res = Rational(0) % Rational(-242);
+            VERIFY_ARE_EQUAL(res, 0);
+            for (auto number : { 343654332, 0, -23423 })
+            {
+                try
+                {
+                    res = Rational(number) % Rational(0);
+                    Assert::Fail();
+                }
+                catch (uint32_t t)
+                {
+                    if (t != CALC_E_INDEFINITE)
+                    {
+                        Assert::Fail();
+                    }
+                }
+                catch (...)
+                {
+                    Assert::Fail();
+                }
+
+                try
+                {
+                    res = Rational(Number(1, number, { 0 }), Number(1, 0, { 2 })) % Rational(Number(1, 0, { 0 }), Number(1, 0, { 23 }));
+                    Assert::Fail();
+                }
+                catch (uint32_t t)
+                {
+                    if (t != CALC_E_INDEFINITE)
+                    {
+                        Assert::Fail();
+                    }
+                }
+                catch (...)
+                {
+                    Assert::Fail();
+                }
+            }
+        }
+
+        TEST_METHOD(TestRemainerRational)
+        {
+            //Test with rational numbers
+            auto res = Rational(Number(1, 0, { 250 }), Number(1, 0, { 100 })) % Rational(89);
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"2.5");
+            res = Rational(Number(1, 0, { 3330 }), Number(1, 0, { 1332 })) % Rational(1);
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"0.5");
+            res = Rational(Number(1, 0, { 12250 }), Number(1, 0, { 100 })) % Rational(10);
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"2.5");
+            res = Rational(Number(-1, 0, { 12250 }), Number(1, 0, { 100 })) % Rational(10);
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"-2.5");
+            res = Rational(Number(-1, 0, { 12250 }), Number(1, 0, { 100 })) % Rational(-10);
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"-2.5");
+            res = Rational(Number(1, 0, { 12250 }), Number(1, 0, { 100 })) % Rational(-10);
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"2.5");
+            res = Rational(Number(1, 0, { 1000 }), Number(1, 0, { 3 })) % Rational(1);
+            VERIFY_ARE_EQUAL(res.ToString(10, FMT_FLOAT, 8), L"0.33333333");
+            res = Rational(Number(1, 0, { 1000 }), Number(1, 0, { 3 })) % Rational(-10);
+            auto sdsdas = res.ToString(10, FMT_FLOAT, 8);
```

Comment:
Doesn't look like sdsdas is used? Artifact from debugging something?

Reply:
:(